From b35210bc2facf2269ad9f676e65071034b4d4228 Mon Sep 17 00:00:00 2001
From: iHDeveloper <git@ihdeveloper.me>
Date: Sun, 19 Dec 2021 20:12:21 +0300
Subject: [PATCH] Upgrade Log4j to 2.17 + Improve logging

This patch is for Spigot 1.8.8

It upgrades the log4j dependency version to 2.17
with fixing the conflicts of the version upgrade.

It also improves the logging operation in `MinecraftServer#sendMessage`.

diff --git a/pom.xml b/pom.xml
index a6641f15..daaaf813 100644
--- a/pom.xml
+++ b/pom.xml
@@ -43,7 +43,7 @@
         <dependency>
             <groupId>org.apache.logging.log4j</groupId>
             <artifactId>log4j-core</artifactId>
-            <version>2.8.1</version>
+            <version>2.17.0</version>
             <scope>compile</scope>
         </dependency>
         <dependency>
@@ -180,7 +180,7 @@
             <plugin>
                 <groupId>org.codehaus.mojo</groupId>
                 <artifactId>animal-sniffer-maven-plugin</artifactId>
-                <version>1.13</version>
+                <version>1.20</version>
                 <executions>
                     <execution>
                         <phase>process-classes</phase>
@@ -200,7 +200,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>2.3</version>
+                <version>3.2.4</version>
                 <executions>
                     <execution>
                         <phase>package</phase>
@@ -245,7 +245,7 @@
             <plugin>
                 <groupId>net.md-5</groupId>
                 <artifactId>specialsource-maven-plugin</artifactId>
-                <version>1.2.1</version>
+                <version>1.2.4</version>
                 <executions>
                     <execution>
                         <phase>package</phase>
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 1d29ae78..f3f093be 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1109,7 +1109,7 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
     }
 
     public void sendMessage(IChatBaseComponent ichatbasecomponent) {
-        MinecraftServer.LOGGER.info(ichatbasecomponent.c());
+        MinecraftServer.LOGGER.info("{}", ichatbasecomponent.c()); // CraftBukkit
     }
 
     public boolean a(int i, String s) {
-- 
2.23.0

